reference/UiTPAS.v2.json

```diff
@@ -334,150 +352,955 @@
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/ReportStatus"
+                },
+                "examples": {
+                  "Example of started export": {
+                    "value": {
+                      "status": "started",
+                      "reportId": "232432"
+                    }
+                  }
+                }
+              }
+            }
+          },
+          "401": {
+            "$ref": "#/components/responses/Unauthorized"
+          },
+          "403": {
+            "$ref": "#/components/responses/Forbidden"
+          }
+        },
+        "tags": [
+          "Financial"
+        ],
+        "description": "Starts a financial report export. The result of this request is a `reportId` that can be used in the /status and /download requests."
+      }
+    },
+    "/financials/organizers/{organizerId}/report/{reportId}/status": {
+      "parameters": [
+        {
+          "schema": {
+            "type": "string"
+          },
+          "name": "organizerId",
+          "in": "path",
+          "required": true
+        },
+        {
+          "schema": {
+            "type": "string"
+          },
+          "name": "reportId",
+          "in": "path",
+          "required": true
+        }
+      ],
+      "get": {
+        "summary": "Get financial report status",
+        "tags": [
+          "Financial"
+        ],
+        "responses": {
+          "200": {
+            "description": "OK",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/ReportStatus"
+                },
+                "examples": {
+                  "Example of a started report": {
+                    "value": {
+                      "reportId": "232432",
+                      "status": "started"
+                    }
+                  },
+                  "Example of an available report": {
+                    "value": {
+                      "reportId": "2332431",
+                      "status": "available"
+                    }
+                  }
+                }
+              }
+            }
+          },
+          "401": {
+            "$ref": "#/components/responses/Unauthorized"
+          },
+          "403": {
+            "$ref": "#/components/responses/Forbidden"
+          },
+          "404": {
+            "description": "Not Found",
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/Error"
+                }
+              }
+            }
+          }
+        },
+        "operationId": "get-financials-organizers-reports-status",
+        "description": "Retrieve the status of a previously started report export."
+      }
+    },
+    "/financials/organizers/{organizerId}/reports/{reportId}/download": {
```

Comment:
Alle endpoints over financiële rapportages bevatten nu een `organizerId`. Gaan er ooit endpoints zijn voor financiële rapportages die je zonder `organizerId` kan aanspreken? Zo niet zou ik eerder volgende structuur voorstellen:

`/organizers/{organizerId}/financial-reports/...`

Bijkomend vermijden we best werkwoorden zoals `download`, de HTTP method zou de actie moeten omschrijven.

Dus ik zou bijvoorbeeld voorstellen:

- `GET /organizers/{organizerId}/financial-reports/periods/` => Voor de mogelijke periodes
- `POST /organizers/{organizerId}/financial-reports/` => Voor het aanmaken
- `GET /organizers/{organizerId}/financial-reports/{reportId}` => Voor de status, en eventueel ook ineens een download URL wanneer het rapport klaar is?
- `GET /organizers/{organizerId}/financial-reports/{reportId}.zip` => Voor de download (kan mee als `url` property in de response data van het vorige endpoint?)